### ISSUE TITLE
Setting progress handler on config for an upload causes OPTIONS preflight to be executed.

### ISSUE THREAD

**Issue by johanmynhardt** [mzabriskie/axios#356]
Hello :)

It took me a while to reproduce this issue in a consistent manner.

**TL;DR**: Setting a progress handler shouldn't cause the request to change in any way.

What I am experiencing though, is that when a progress handler is set, an `OPTIONS` preflight is sent, which differs from when it's not set. It appears to be related to some form of caching as when (in Chrome) I use the developer console and do not disable the cache, it's only occurring sometimes. I could not figure out how to consistently predict when it will happen.
As soon as caching is disabled, the behaviour is consistently occurring.

To illustrate, I have set up a CodePen at http://codepen.io/johanmynhardt/full/VjmoqR/

In short, it has 2 file inputs, one handling a file selection without a progress handler, and the other with the handler set.

### Without progress listener
When you keep an eye on the XHR requests in Chrome's Network tab after selecting a file from the file selection dialog (using first file select input), it will trigger a single `POST` upload to `httpbin` (https://httpbin.org/post) - Happy days.

### With an added progress listener
When a file is now selected using the 2nd file select input, 2 request entries will be logged in the Network tab, of which the first is an `OPTIONS` request, and the 2nd is a `POST` as expected.

I don't know if I'm missing something or whether is a bug.

**Comment by mzabriskie** [comment]
I did a quick search and can't find anything in the spec that suggests why a preflight requests is used for requests using `onprogress`. Using Firefox the XHR that doesn't use `onprogress` makes a simple `POST`. The XHR that does use `onprogress` makes an `OPTIONS` preflight request followed by a `POST` request that is identical to the non-progress request.

So I can't really explain _why_ it's making the preflight request, but axios is only adding an event listener to the request, nothing more, so I don't know how it can be avoided. https://github.com/mzabriskie/axios/blob/master/lib/adapters/xhr.js#L146-L152

**Comment by johanmynhardt** [comment]
OK, thank you for having a look Matt, I appreciate it! I will do some more
digging in the morning :)

**Comment by johanmynhardt** [comment]
_Sigh..._ I've done some more careful reading, and it seems easy to get it wrong, especially since XMLHttpRequest and CORS spec is scattered around a bit (IMO).

With much concentration in reading the following and traverse the web of links, it is clear that I missed something:
https://dvcs.w3.org/hg/xhr/raw-file/tip/Overview.html#the-send()-method (4.6.6, item 10, Otherwise-part)
https://www.w3.org/TR/cors/#cross-origin-request

I don't feel bad for expecting what I did :smiley:
In the end it does what is expected.
